lib/domain/entities/activity.dart

```diff
@@ -0,0 +1,57 @@
+import 'package:freezed_annotation/freezed_annotation.dart';
+import 'package:movna/domain/entities/sport.dart';
+import 'package:movna/domain/entities/track_point.dart';
+import 'package:movna/domain/entities/track_segment.dart';
+
+part 'activity.freezed.dart';
+
+/// Training activity session.
+@freezed
+class Activity with _$Activity {
+  const Activity._();
+
+  const factory Activity({
+    /// The time at which this activity begun.
+    required DateTime startTime,
+
+    /// The time at which this activity ended.
+    /// Null when activity is not ended yet.
+    DateTime? stopTime,
+
+    /// The sport practiced during this activity. Can be unspecified.
```

Comment:
prefer using null rather than unspecified

for example the field averageSpeedInMetersPerSecond can be unspecified but it cannot be null